packages/utils/social-feed.ts

```diff
@@ -68,19 +72,38 @@ export const hashtagMatch = (text: string) =>
 export const urlMatch = (text: string) =>
   text.match(new RegExp(URL_REGEX, "g"));
 
-export const postResultToPost = (networkId: string, postResult: PostResult) => {
-  return {
-    category: postResult.category,
-    isDeleted: postResult.deleted,
-    identifier: postResult.identifier,
-    metadata: postResult.metadata,
-    parentPostIdentifier: postResult.parent_post_identifier,
-    subPostLength: postResult.sub_post_length,
-    reactions: postResult.reactions,
-    // TODO: We need to parse, because we're using Post[] from social-feed API and PostResult[] from social-feed contract
-    createdBy: getUserId(networkId, postResult.post_by),
-    createdAt: JSON.parse(postResult.metadata).createdAt,
-  } as Post;
+export const postResultToPost = (
+  networkId: string,
+  postResult: PostResultExtra | PostResult
+) => {
+  if ("isInLocal" in postResult) {
+    return {
+      category: postResult.category,
+      isDeleted: postResult.deleted,
+      identifier: postResult.identifier,
+      metadata: postResult.metadata,
+      parentPostIdentifier: postResult.parent_post_identifier,
+      subPostLength: postResult.sub_post_length,
+      reactions: postResult.reactions,
+      // TODO: We need to parse, because we're using Post[] from social-feed API and PostResult[] from social-feed contract
+      createdBy: getUserId(networkId, postResult.post_by),
+      createdAt: JSON.parse(postResult.metadata).createdAt,
+      isInLocal: postResult.isInLocal,
+    } as PostExtra;
+  } else {
+    return {
+      category: postResult.category,
+      isDeleted: postResult.deleted,
+      identifier: postResult.identifier,
+      metadata: postResult.metadata,
+      parentPostIdentifier: postResult.parent_post_identifier,
+      subPostLength: postResult.sub_post_length,
+      reactions: postResult.reactions,
+      // TODO: We need to parse, because we're using Post[] from social-feed API and PostResult[] from social-feed contract
+      createdBy: getUserId(networkId, postResult.post_by),
+      createdAt: JSON.parse(postResult.metadata).createdAt,
+    } as Post;
+  }
```

Comment:
We could shortcut this:

```
let post = {
      category: postResult.category,
      isDeleted: postResult.deleted,
      identifier: postResult.identifier,
      metadata: postResult.metadata,
      parentPostIdentifier: postResult.parent_post_identifier,
      subPostLength: postResult.sub_post_length,
      reactions: postResult.reactions,
      // TODO: We need to parse, because we're using Post[] from social-feed API and PostResult[] from social-feed contract
      createdBy: getUserId(networkId, postResult.post_by),
      createdAt: JSON.parse(postResult.metadata).createdAt,
    } as Post

if ("isInLocal" in postResult) { 
  post  = {...post, isInLocal: postResult.isInLocal}  as PostExtra;
}

something like that.

Btw, do we need to check the value of `isInLocal` or just check if property exists ?


```